authorization/handler.go

```diff
@@ -16,9 +16,12 @@ import (
 
 type AuthorizerClient interface {
 	IsAuthorizedWithContext(ctx context.Context, userID, action string, resource *proto.Origin) (bool, error)
+	IsAuthorizedWithReasonWithContext(ctx context.Context, userID, action string, resource *proto.Origin) (bool, string, error)
 	GetResourceWithContext(ctx context.Context, id, originType string) (proto.Origin, error)
 }
 
+const resourceNotFoundReason = "resource_not_found"
```

Comment:
```suggestion
const authorizerResourceNotFoundReason = "resource_not_found"
```
maybe? or perhaps a comment would be better to explain why it's here, might be confusing in a few months.